api/apps/api/src/modules/cost-surface/cost-surface.utils.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * @note: I'd like to add this method in the Project class, but then it will mess a lot with all the object literals
+ *        instantiated in tests. Better ideas welcome
+ *
+ */
+import { Project } from '@marxan-api/modules/projects/project.api.entity';
+
+export const getProjectDefaultCostSurfaceId = (project: Project): string => {
+  if (project.costSurfaces?.length === 1) return project.costSurfaces[0].id;
+  const defaultCostSurface = project.costSurfaces?.find(
+    (costSurface) => costSurface.isDefault,
+  );
+  if (!defaultCostSurface) {
+    throw new Error(
+      `Could not get default Cost Surface for Project: ${project.id}`,
+    );
+  }
+  return defaultCostSurface.id;
```

Comment:
For the first comment: Yes, you are totally right, does not make any sense to not include any handling in the very beggining

For the second comment: Yes, it's exactly because any literal that is expected to be a project, also expect the method. This is one of the very few moments when I miss Java.

I also like better the idea of having a static method (which I already included), rather than having a new file with a utility function that probably nobody will remember in 6 months.

However, I feel like this really should me a method available for the instance since it is actually interacting with the instance itself

Reply:
about where to place this method, tbh, I wouldn't bother with the Project class at all. you ask 10 people about these things, you get 20 opinions, so just my 2c, but I do prefer small, clearly bounded (in terms of domain role) classes, even if they just have one method because that's what that trait of the domain is supposed to be doing. I see no compelling rationale for bundling up lots of behaviour in ever-expanding megaclasses.

Reply:
I will create a minimal utility function for this, since, as simple as it might seem, if we follow the port/infra/application pattern, it would include several files and several layers to just find a element in a array, IMO

Reply:
yep, to clarify that's all I was suggesting, in practice, a standalone function or a class method in a class with just that, used mostly as namespacing - absolutely no need to introduce port/adaptor madness for this!